docs/docs/higher-order-components.md

```diff
@@ -0,0 +1,491 @@
+// Now apply an HOC
+const EnhancedComponent = enhance(WrappedComponent);
+
+// The enhanced component has no static method
+typeof EnhancedComponent.staticMethod === 'undefined' // true
+```
+
+To solve this, you could copy the methods onto the container before returning it
+
+```js
+function enhance(WrappedComponent) {
+  class Enhance extends React.Component {/*...*/}
+  // Must know exactly which method(s) to copy :(
+  Enhance.staticMethod = WrappedComponent.staticMethod;
+  return Enhance;
+}
+```
+
+but this requires you to know exactly which methods need to be copied. You can use [hoist-react-non-statics](https://github.com/mridgway/hoist-non-react-statics) to automatically copy all non-React static methods:
+
+```js
+import hoistNonReactStatic from 'hoist-non-react-statics';
+function enhance(WrappedComponent) {
+  class Enhance extends React.Component {/*...*/}
+  hoistNonReactStatic(Enhance, WrappedComponent);
+  return Enhance;
+}
+```
+
+Another possible solution is to export the static method separately from the component itself.
+
+```js
+// Instead of...
+MyComponent.someFunction = someFunction;
+export default MyComponent;
+
+// ...export the method separately...
+export { someFunction };
+
+// ...and in the consuming module, import both
+import MyComponent, { someFunction } from './MyComponent.js';
+```
+
+### Refs Aren't Passed Through
+
+If you add a ref to an element whose component is the result of applying an HOC, the ref refers to the container component, not the original, wrapped component:
+
+```js
+// A higher-order component
+function enhance(WrappedComponent) {
+  class Enhance extends React.Component {
+    render() {
+      return <WrappedComponent {...this.props} />;
+    }
+  }
+  return Enhance;
+}
+
+const EnhancedComponent = enhance(WrappedComponent);
+
+// Inside a render method
+<EnhancedComponent
+  ref={(instance) => {
+    // instance is an instance of Enhance, not WrappedComponent!
+  }}
+/>
+```
+
+This is because `ref` is not really a prop—like `key`, it's handled specially by React, so it's not possible to pass a ref from one component to the next.
+
+If you find yourself facing this problem, the ideal solution is to figure out how to avoid using `ref` at all. Occasionally, users who are new to the React paradigm rely on refs in situations where a prop would work better.
+
+That said, there are times when `refs` are a necessary escape hatch—React wouldn't support them otherwise.
+
+In that case, one solution is to pass a ref callback as a prop with a different name (e.g. `innerRef`, though the specific name is not important) and manually call it during mount and unmount.
+
+```js
+class WrappedComponent extends React.Component {
+  componentDidMount() {
+    this.props.innerRef(this);
+  }
+  componentWillUnmount() {
+    this.props.innerRef(null);
+  }
+  render() {
+    const { innerRef, ...rest } = this.props;
+    return <WrappedComponent {...rest} />;
```

Comment:
I'm confused by this, isn't `WrappedComponent` the same class as the one we're inside of?

In general I would prefer a more specific example here. Like how would you expose/call an imperative `focus()` method on a DOM node of a connected component.

```js
<div ref={node => this.props.registerFocus(node ? () => node.focus() : null)}>
  <Stuff />
</div>
```

```js
<ConnectedField registerFocus={focus => this.focusField = focus} />

// later
this.focusField();
```